source/common/config/grpc_mux_impl.cc

```diff
@@ -321,7 +380,17 @@ void GrpcMuxImpl::onEstablishmentFailure() {
       watch->callbacks_.onConfigUpdateFailed(
           Envoy::Config::ConfigUpdateFailureReason::ConnectionFailure, nullptr);
     }
+    if (starting_) {
+      // On the initialization of the gRPC mux, if connection to the xDS server fails, load the
+      // persisted config, if available. The locally persisted config will be used until
+      // connectivity is established with the xDS server.
```

Comment:
Can you add an integration tests that emulates the "connectivity is re-established" scenario?

Reply:
Added a test for this